### PR TITLE
Port to use the swift2 branch from ReactiveCocoa

### PR DESCRIPTION
**Pull request by liscio**
These are all the changes required to get this framework compiling against RAC4 as it exists in the `swift2` branch.

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1 +1 @@
-github "ReactiveCocoa/ReactiveCocoa" "v3.0-beta.8"
+github "ReactiveCocoa/ReactiveCocoa" "swift2"
```
Comment (neilpa) [inline Cartfile]: Can you target the `v4.0-alpha.1` tag or a specific commit instead. Pinning to branch heads is super fragile for consumers - see neilpa/Rex#31 where I learned my lesson.

**Comment by liscio** [comment]
@neilpa Fixed, as requested! :)

**Comment by neilpa** [comment]
We should leave this open until the Swift 2 branch of RAC 4 is merged into master.

**Comment by neilpa** [comment]
I made a swift2 branch if you want to open a new PR against it.

**Pull request closed without merging**